### PR TITLE
fix(vcDoubleClick): don't require double-click on active voice channel, fix stage channels patch

### PR DESCRIPTION
**Pull request by D3SOX**

### CODE REVIEW

File: src/plugins/vcDoubleClick.ts [hunk 1]
```diff
@@ -63,6 +64,13 @@ export default definePlugin({
     schedule(cb: () => void, e: any) {
         // support from stage and voice channels patch
         const id = e?.id ?? e.props.channel.id as string;
+        const activeVoiceChannel = SelectedChannelStore.getVoiceChannelId();
+        if (activeVoiceChannel) {
```
Comment (Vendicated) [inline src/plugins/vcDoubleClick.ts:68]:
```suggestion
        if (SelectedChannelStore.getVoiceChannelId() === id) {
```

Comment (D3SOX) [reply src/plugins/vcDoubleClick.ts:68]: Why? The suggestion doesn't make sense, what did you want to achieve?

Comment (Vendicated) [reply src/plugins/vcDoubleClick.ts:68]: the outer if is unnecessary, you can just do the comparison right away. now you can also eliminate the variable

Comment (D3SOX) [reply src/plugins/vcDoubleClick.ts:68]: I just wanted to be sure that it's not undefined and `id` is also `undefined` and then it's equal but yeah

Comment (Vendicated) [reply src/plugins/vcDoubleClick.ts]: Id shouldn't ever be undefined, should it? in any case, I would just add a separate check that returns early if id is undefined, that's way prettier

**Comment by D3SOX** [comment]
For the stage channels patch: I'm not sure if this is different in Canary and Stable right now. I can't test it with Canary rn but maybe it should cover both ways

**Comment by Vendicated** [comment]
special canary treatment is kinda pointless and will be very cumbersome, I would just make sure it works on stable

**Comment by D3SOX** [comment]
As of this PR, the patch works in stable

File: src/plugins/vcDoubleClick.ts [hunk 2]
```diff
@@ -46,8 +47,8 @@ export default definePlugin({
                 },
                 // stage channels
                 {
-                    match: /onClick:(\w+)\?void 0:this\.handleClick,/g,
-                    replace: "onClick:$1?void 0:(...args)=>Vencord.Plugins.plugins.vcDoubleClick.schedule(()=>{this.handleClick(...args);}, args[0]),",
+                    match: /onClick:(.*)this\.handleClick,/g,
```
Comment (Vendicated) [inline src/plugins/vcDoubleClick.ts:50]: This greedy operator is dangerous, do something else or limit it to as litte chars as possible, such as `.{0,20}`

**Review by Vendicated — CHANGES_REQUESTED** [review]

File: src/plugins/vcDoubleClick.ts (hunk 2 above)
Comment (D3SOX) [reply src/plugins/vcDoubleClick.ts]: Updated

File: src/plugins/vcDoubleClick.ts [hunk 3]
```diff
@@ -63,6 +64,10 @@ export default definePlugin({
     schedule(cb: () => void, e: any) {
         // support from stage and voice channels patch
         const id = e?.id ?? e.props.channel.id as string;
+        if (SelectedChannelStore.getVoiceChannelId() === id) {
+            cb();
```
Comment (Vendicated) [inline src/plugins/vcDoubleClick.ts:68]: nitpick but u can just `return void cb()`

Comment (D3SOX) [reply src/plugins/vcDoubleClick.ts:68]: Didn't know about that syntax and got this warning by SonarLint lol

image

**Pull request merged**